storage_controller/src/scheduler.rs

```diff
@@ -383,39 +416,49 @@ impl Scheduler {
     /// the same tenant on the same node.  This is a soft constraint: the context will never
     /// cause us to fail to schedule a shard.
     pub(crate) fn schedule_shard(
-        &self,
+        &mut self,
```

Comment:
why mut: we seem to operate on copies. Perhaps I'm missing the mutation

Reply:
It's where we do `cached_score`: the utilization object needs to be mut.  This could be refactored in future to make that into interior mutability with an atomic, I don't really love carrying mut as much as we are doing here.